crates/api_common/src/utils.rs

```diff
@@ -78,6 +78,21 @@ pub async fn is_mod_or_admin_opt(
   }
 }
 
+#[tracing::instrument(skip_all)]
+pub async fn is_mod_of_any_or_admin(
+  pool: &mut DbPool<'_>,
+  person: &Person,
+) -> Result<(), LemmyError> {
+  check_user_valid(person)?;
+
+  let is_mod_of_any_or_admin = CommunityView::is_mod_of_any_or_admin(pool, person.id).await?;
+  if !is_mod_of_any_or_admin {
+    Err(LemmyErrorType::NotAModOrAdmin)?
+  } else {
+    Ok(())
+  }
+}
```

Comment:
This can also be merged into the function below.

Reply:
I've removed the unecessary one now.